sdk/src/Services/S3/Custom/Transfer/Internal/_async/MultipartUploadCommand.async.cs

```diff
@@ -240,22 +238,16 @@ private async Task UploadNonSeekableStreamAsync(TransferUtilityUploadRequest req
                                 bool isLastPart = readBytesCount == 0;
                                 var partSize = nextUploadBuffer.Position;
                                 nextUploadBuffer.Position = 0;
-                                var partResponse = await _s3Client.UploadPartAsync(new UploadPartRequest()
-                                {
-                                    BucketName = request.BucketName,
-                                    Key = request.Key,
-                                    UploadId = initiateResponse.UploadId,
-                                    InputStream = nextUploadBuffer,
-                                    PartSize = partSize,
-                                    PartNumber = partNumber,
-                                    IsLastPart = isLastPart
-                                }).ConfigureAwait(false);
+                                UploadPartRequest uploadPartRequest = ConstructUploadPartRequestForNonSeekableStream(nextUploadBuffer, partNumber, partSize, isLastPart, initiateResponse);
+
+                                var partResponse = await _s3Client.UploadPartAsync(uploadPartRequest).ConfigureAwait(false);
                                 Logger.DebugFormat($"Uploaded part {partNumber}. (Last part = {isLastPart}, Part size = {partSize}, Upload Id: {initiateResponse.UploadId}");
-                                partETags.Add(new PartETag { PartNumber = partResponse.PartNumber, ETag = partResponse.ETag });
+                                uploadPartResponses.Add(partResponse);
                                 partNumber++;
                                 nextUploadBuffer = new MemoryStream(partBuffer);
```

Comment:
We are creating a new `MemoryStream` without disposing the previous `MemoryStream`.

Reply:
Fixed. Please review.